airflow/providers/amazon/aws/operators/ec2_start_instance.py

```diff
@@ -0,0 +1,88 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+import time
+from typing import Optional
+
+from airflow.models import BaseOperator
+from airflow.providers.amazon.aws.hooks.ec2 import EC2Hook
+from airflow.utils.decorators import apply_defaults
+
+
+class EC2StartInstanceOperator(BaseOperator):
+    """
+    Start AWS EC2 instance using boto3.
+
+    :param instance_id: id of the AWS EC2 instance
+    :type instance_id: str
+    :param aws_conn_id: aws connection to use
+    :type aws_conn_id: str
+    :param region_name: (optional) aws region name associated with the client
+    :type region_name: Optional[str]
+    :param check_interval: time in seconds that the job should wait in
+        between each instance state checks until operation is completed
+    :type check_interval: float
+    """
+
+    template_fields = ["region_name"]
+    ui_color = "#eeaa11"
+    ui_fgcolor = "#ffffff"
+
+    @apply_defaults
+    def __init__(self,
+                 instance_id: str,
+                 aws_conn_id: str = "aws_default",
+                 region_name: Optional[str] = None,
+                 check_interval: float = 15,
+                 *args,
+                 **kwargs):
+        super().__init__(*args, **kwargs)
+        self.instance_id = instance_id
+        self.aws_conn_id = aws_conn_id
+        self.region_name = region_name
+        self.check_interval = check_interval
+
+    def execute(self, context):
+        ec2_hook = self.get_hook()
+        self.log.info("Starting EC2 instance %s", self.instance_id)
+
+        instance = ec2_hook.get_instance(instance_id=self.instance_id)
+        instance.start()
+        instance_state = ec2_hook.get_instance_state(
+            instance_id=self.instance_id
+        )
+
+        while instance_state != "running":
+            self.log.info("instance state: %s", instance_state)
+            time.sleep(self.check_interval)
+            instance_state = ec2_hook.get_instance_state(
+                instance_id=self.instance_id
+            )
```

Comment:
This method seems to be used in both operators. Can we extract it to hook as something like `wait_for_status` ?

Reply:
Sure, I will.